vocabulary/javascript/debugging-try-and-catch.md

```diff
@@ -0,0 +1,35 @@
+# Try and catch
+
+## Errors Will Happen
+
+If your program keeps crashing at a particular spot, try surrounding that code
+with try catch blocks to investigate the error object that gets returned.
+
+The `try` statement allows you to define a block of code to be tested for errors
```

Comment:
You could use the `try-catch` statements for finding the errors, but if you don't. The errors will be logged automatically (with `console.error)` to the console by the browser. So it's not really that useful for that use case.

`try-catch`-es are usually used around things that can fail such as getting data from a server (the network could be disconnected, the server can be down, timeouts,...). You could then show a message to the user what went wrong.

errors you get when developing `!==` errors users get when using your software.

Reply:
@sannesofie The most important thing is that you understood what I said. If that's not the case, let me know and I'll try and explain again. If you did understand, try changing line 5-6 to the thing you learned from the explanation. 🙂

Reply:
@arnochauveau I thought I understood it but I guess I did not LOL

Reply:
Let's plan a call next week to go over it again 😁

Reply:
Thanks that would be great ;) I will message you on Discord